feat(website): getting started FAQs (#777)
twilio-labs/paste: packages/paste-website/src/pages/getting-started/faqs/index.mdx

description: Frequently asked questions when getting started with Paste.
slug: /getting-started/faqs/
---

import {graphql} from 'gatsby';
import {Stack} from '@twilio-paste/stack';
import {Paragraph} from '@twilio-paste/paragraph';
import {UnorderedList, ListItem} from '@twilio-paste/list';
import {Anchor} from '@twilio-paste/anchor';
import {Disclosure, DisclosureHeading, DisclosureContent} from '@twilio-paste/disclosure';
import {InlineCode} from '../../../components/Typography';

export const pageQuery = graphql`
  {
    mdx(fields: {slug: {eq: "/getting-started/faqs/"}}) {
      fileAbsolutePath
      frontmatter {
        slug
        title
      }
      headings {
        depth
        value
      }
    }
  }
`;

<content>

<h1>{props.pageContext.frontmatter.title}</h1>

<Paragraph>{props.pageContext.frontmatter.description}</Paragraph>

</content>

---

<contentwrapper>

<PageAside data={props.data.mdx} />

<content>

## Engineering

<Stack orientation="vertical" spacing="space140">
  <Disclosure visible>
    <DisclosureHeading as="h3" variant="heading30">
      What browser support does Paste have?
    </DisclosureHeading>
    <DisclosureContent>
      <Paragraph>

>         We support the latest and latest minus one versions of all modern browsers on the market today. Most notible

Author: ```suggestion
        We support the latest and latest minus one versions of all modern browsers on the market today. Most notable
```

      </Paragraph>
      <UnorderedList>
        <ListItem>Google Chrome</ListItem>
        <ListItem>Safari</ListItem>
        <ListItem>Firefox</ListItem>
        <ListItem>MS Edge</ListItem>
        <ListItem>Chromium Edge</ListItem>
      </UnorderedList>
    </DisclosureContent>
  </Disclosure>
  <Disclosure>
    <DisclosureHeading as="h3" variant="heading30">
      Paste components don't support a className property, why is that?
    </DisclosureHeading>
    <DisclosureContent>
      <Paragraph>
        Whilst we are acutely aware that a Design System cannot do and hold everything that is needed for each
        individual product, and that extension and customization is the key to a successful Design System at scale, we
        will not be providing that functionality via a custom classname.
      </Paragraph>
      <Paragraph>We won't be doing this for a number of reasons:</Paragraph>
      <UnorderedList>
        <ListItem>
          Global classnames, especially overrides, impact the confidence teams can have on the state of their styling.
          There is no guarantee you won't affect someone else or they won't affect you, impacting the success of

>           independent teams and confidence in your deploys. This particularly applies to a number of Twilio products

Author: ```suggestion
          independent teams and confidence in your deployments. This particularly applies to a number of Twilio products
```

          that are multi-tenant UI, where multiple independent teams are providing UI to a single page
        </ListItem>
        <ListItem>Uncontrolled customization causes design fragmentation</ListItem>
        <ListItem>
          Uncontrolled customization impacts the ability of the Design System team to push changes to and evolve the
          Design System, without adversely impacting downstream consumers
        </ListItem>
      </UnorderedList>
      <Paragraph>
        Above all, custom classnames erode trust in the system, the components, the product and adversely affect the
        trust of the customer.
      </Paragraph>
      <Paragraph>
        Customization and extension will be provided, but it will be provided via a controlled API, within the
        constraints of the Design System. More on that soon.
      </Paragraph>
    </DisclosureContent>
  </Disclosure>
  <Disclosure>
    <DisclosureHeading as="h3" variant="heading30">
      Why does Paste use React-uid to generate ids? Why don't you use static ids?
    </DisclosureHeading>
    <DisclosureContent>
      <Paragraph>

>         ID references in a webpage must be unique for them to work correctly, especially when supporting assistive

Author: ```suggestion
        ID references in a web page must be unique for them to work correctly, especially when supporting assistive
```

        technologies such as screen readers. In a multi tenant, single page application you do not always have control
        of what is loaded into a webpage at any one time. As a result static ids can clash and cause duplicates.
      </Paragraph>
      <Paragraph>

>         To guarantee that every id in a webpage is unique, regardless of what was loaded on the page and in what order,

Author: ```suggestion
        To guarantee that every id in a web page is unique, regardless of what was loaded on the page and in what order,
```

        we use react-uid to generate ids for us.
      </Paragraph>
    </DisclosureContent>
  </Disclosure>
  <Disclosure>
    <DisclosureHeading as="h3" variant="heading30">
      OK, so you don't support classNames or static IDs, how do I create DOM references for my tests?
    </DisclosureHeading>
    <DisclosureContent>

>       <Paragraph>We recommend using data attributes for your test indentifiers.</Paragraph>

Author: ```suggestion
      <Paragraph>We recommend using data attributes for your test identifiers.</Paragraph>
```

        separately. We only run them through Chrome currently.
      </Paragraph>
      <Paragraph>
        We also use Chromatic for UI review where any visual changes are reviewed by a designer for all components.
      </Paragraph>
      <Paragraph>All these tests are run on every commit and are required checks on all pull requests.</Paragraph>
    </DisclosureContent>
  </Disclosure>
  <Disclosure>
    <DisclosureHeading as="h3" variant="heading30">
      How do I get the fonts to load in my application?
    </DisclosureHeading>
    <DisclosureContent>
      <Paragraph>
        Paste leaves loading the font files up to the consumer of the system, to accomodate any technical or licensing
        restrictions a product might have.
      </Paragraph>
      <Paragraph>
        For the default theme Paste uses Inter. This is available as a{' '}
        <Anchor href="https://fonts.google.com/specimen/Inter">Google web font</Anchor> can be loaded in a number of
        ways into your application. Please choose the one that best suits your preferred choice of asset delivery, or

>         follow how{' '}

Author: ```suggestion
        follow{' '}
```

        <Anchor href="https://github.com/twilio-labs/paste/blob/main/packages/paste-website/src/assets/css/fonts.css#L105">
          how we load the font
        </Anchor>{' '}
        on this website.
      </Paragraph>
      <Paragraph>

>         For the Console theme, due to licensing restictions only web properties under certain domains are allowed to

Author: ```suggestion
        For the Console theme, due to licensing restrictions only web properties under certain domains are allowed to
```
